### DISCUSSION TITLE
Stopping automatic GitHub Action on push

### DISCUSSION THREAD

**Discussion by DouglasTitze** [community/community#50868] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello, I currently use the [chirpy ](https://github.com/cotes2020/jekyll-theme-chirpy) Jekyll theme on my website. When I duplicated the template, it came with a GitHub Action to push and deploy the website, which was helpful, considering I do not know how to do this. Now it has become an issue.
Is there a way to keep the GitHub action inside GitHub, so I can build and deploy when I like but disable it from automatically performing when I push to GitHub?

GitHub [workflow](https://github.com/DouglasTitze/DouglasTitze.github.io/blob/main/.github/workflows/pages-deploy.yml) from the template, which I am referring to.

**Comment by Anmol-Baranwal** [comment, marked as answer]
Yes, you can disable the GitHub Action from automatically deploying your website when you push to GitHub by commenting out or removing the following lines from the workflow file:

```yaml
on:
  push:
    branches:
      - main
      - master
```

These lines define the trigger for the workflow, which is set to run on every push to the `main` and `master` branches. By removing these lines, the workflow will not be triggered automatically on push.

<hr> 

You can still manually trigger the workflow from the Actions tab in your repository to build and deploy your website. To do so, follow these steps:

1. Go to the "Actions" tab in your repository on GitHub.
2. Click on the name of the workflow you want to run (in this case, "Build and Deploy").
3. Click on the "Run workflow" button.
4. Select the branch you want to deploy (e.g. main).
5. Click on the "Run workflow" button to start the workflow.

This will trigger the workflow to build and deploy your website manually, without the need for automatic deployment on push.

**Reply by DouglasTitze** [reply]
Thank you for your help! I prevented "Build and Deploy" from occurring when I push while keeping it in my GitHub Actions by doing the following.
```yml
on: 
  workflow_dispatch:
    paths-ignore:
      - .gitignore
      - README.md
      - LICENSE
```
Unfortunately, "pages build and deployment" still runs anytime I commit. Is there a way to change this also?

**Reply by Anmol-Baranwal** [reply]
if you remove the `workflow_dispatch` event from the `on` section of your YAML file, the workflow will not be triggered when you push commits to your repository. Instead, it will only run when you manually trigger it from the Actions tab on GitHub.

<hr>

To prevent the "pages build and deployment" job from running on every commit, you can modify the `on` section to only trigger the workflow when specific conditions are met.

For example, you could modify the `on` section to only trigger the workflow when a new release is created, like this:

```yaml
on:
  release:
    types: [created]
```
This YAML code specifies that the workflow should only run when a new release is created. When a new release is created, GitHub automatically generates a tag that you can use to deploy the website.

<hr>

Alternatively, if you don't want to create a new release every time you want to deploy the website, you can create a custom event that triggers the workflow. For example, you could create a custom event called `deploy` and trigger the workflow with the following YAML code:

```yaml
on:
  repository_dispatch:
    types: [deploy]
```

Then, whenever you want to deploy the website, you can trigger the workflow manually by clicking on the "Run workflow" button in the Actions tab and selecting the `deploy` event.

<hr>

You can also add additional conditions to the `on` section to further control when the workflow should be triggered. For example, you could specify that the workflow should only be triggered when a commit message includes a specific keyword or when a specific tag is pushed.

Here's an example of how you could modify the `on` section of the workflow YAML file to trigger the workflow when a specific tag is pushed:

```yaml
on:
  push:
    tags:
      - v*
```

This updated YAML code specifies that the workflow should only run when a tag starting with "v" is pushed.

**Reply by MavisMiniatures** [reply]
I have the same issue but the workflow file does not exist.
